### ISSUE TITLE
model architecture

### ISSUE THREAD

**Issue by arunvellat** [thennal10/l3prune#1]
It appears the decoder only LLM is appended with a weighted pooling layer and supervised contrastive fine-tuning is performed.  Is the bidirectional attention mechanism, or masked next token prediction, or unsupervised contrastive learning from LLM2Vec incorporated here ?

**Comment by thennal10** [comment]
Not as of yet! For the sake of keeping the codebase clean and tested I decided not to necessarily include all of the LLM2Vec steps. In my earlier experiments the benefits of the full LLM2Vec pipeline was extremely minimal, and bidirectional attention in particular requires you to modify the arch of every new model. Might consider implementing something a bit more universal like NVEmbed's methodology (remove causal attention mask + latent attention pooling).
